config/config.schema.json

```
@@ -3,6 +3,20 @@
   "title": "Generated schema for Root",
   "type": "object",
   "definitions": {
+    "postgresName": {
+      "type": "string",
+      "minLength": 3,
+      "maxLength": 44,
+      "pattern": "^[a-z0-9][a-z0-9-]*[a-z0-9]$",
+      "description": "Server name must only contain lowercase letters, numbers, and hyphens. The server name must not start or end in a hyphen. Name must be between 3 be 63 characters long but we limited to 44 characters so we have a future option to use NSP: https://learn.microsoft.com/en-us/azure/private-link/network-security-perimeter-concepts"
+    },
+    "keyVaultName": {
+      "type": "string",
+      "minLength": 3,
+      "maxLength": 24,
+      "pattern": "^[a-z](?:[a-z0-9]*(-[a-z0-9]+))*$",
```

I find this more readable:

```suggestion
      "pattern": "^[a-z][a-z0-9]*(-[a-z0-9]+)*$",
```